09 Redux/src/components/members/membersPage.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { connect } from 'react-redux'
+import {Link} from 'react-router';
+import memberEntity from '../../api/memberEntity';
+import MemberAPI from '../../api/memberAPI';
+import MemberRow from './memberRow'
+import loadMembers from '../../actions/loadMembers'
+
+// Presentational
+
+// extends React.Props<MembersPage>
+interface Props extends React.Props<MembersPage>{
+  members? : Array<any>;
+  loadMembers? : () => void;
+}
+
+// StateLessComponent complaining
+// TODO: Check if it's a problem of type definition
+class MembersPage extends React.Component<Props, {}> {
+   context: any;
+
+   constructor(props, context) {
+     super(props, context);
+   }
+
+   static contextTypes = {
+        store: React.PropTypes.object
+   }
+   private unsubscribe: Function;
+
+
+   // Standard react lifecycle function:
+   // https://facebook.github.io/react/docs/component-specs.html
+   public componentDidMount() {
+     this.unsubscribe = this.context.store.subscribe(() => this.forceUpdate());
+     this.props.loadMembers();
+   }
+
+   componentWillUnmount() {
+     this.unsubscribe();
+   }
+
+
+   public render() {
+     if(!this.props.members)
+        return (<div>No data</div>)
+
+
+       return (
+        <div className="row">
+          <h2> Members Page</h2>
+          <Link to="/member">New Member</Link>
+          <table className="table">
+            <thead>
+              <th>
```

Comment:
Done, this is something we have to open an issue and propagate the change to all samples